platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -160,14 +166,13 @@ public class HalfDiskHashMap implements AutoCloseable, Snapshotable, FileStatist
     /** Fork-join pool for HDHM.endWriting() */
     private static volatile ForkJoinPool flushingPool = null;
 
-    private static ForkJoinPool getFlushingPool() {
+    private ForkJoinPool getFlushingPool() {
```

Comment:
This doesn't look nice, to initialize a static field from a non-static method (even with proper synchronization). One solution could be to move this fork-join pool to `MerkleDb` and explicitly make it one per `MerkleDb` instance. It would be slightly differently than static (one per JVM), but should be acceptable.

Reply:
I think to revert such cases to not overcomplicate this PR.

Reply:
Addressed, see [this commit](https://github.com/hashgraph/hedera-services/pull/15562/commits/054cf61cd11bdacfa04bef672823f20bdd858eb5)